src/schema/v2/artist/index.ts

```diff
@@ -627,6 +627,24 @@ export const ArtistType = new GraphQLObjectType<any, ResolverContext>({
                 return headers["x-total-count"] || 0
               },
             },
+            myCollectedArtworks: {
+              type: new GraphQLNonNull(GraphQLInt),
+              resolve: async ({ _id }, _, { meMyCollectionArtworksLoader }) => {
+                if (!meMyCollectionArtworksLoader) {
+                  return 0
+                }
+
+                const { headers } = await meMyCollectionArtworksLoader({
```

Comment:
how about making it more error-proof if the request fails?

Reply:
In what way: you wouldn't want to return an erroneous count (0). You want it to error and return the error.

Reply:
true but the client consuming it will have the entire query request broken, isn't it?

Reply:
Depends on how one handles the error. An error in the field would null the parent like so: 
![](https://capture.static.damonzucconi.com/Screen-Shot-2024-04-18-09-35-04.71-8slc7CTo56ObcfTNjQzmq0zcrV545sHuBLMgLiITCeVRnkdAFKTeB5JutQPowBU0uHW03cFZOzuRiPheqiHdddluqRlifIqRIclC.png)

Either way: IMO you don't want to swallow the error and return null on the field. You want to surface the error for the client to deal with.

I'd rather the page error than have to include error handling code on every field I display on the front end.

Reply:
agreed, handling the errors in the client sounds more real IMO than fake nulls.